web/js/codeworld_shared.js

```diff
@@ -1025,24 +1045,27 @@ function saveProjectAsBase(successFunc) {
             }
 
             $('#directoryTree').tree('selectNode', node);
+
             updateChildrenIndexes(parent);
+
             successFunc(result.value);
         }
         if (result && result.value) {
             saveProjectBase(
                 getNearestDirectory(),
                 result.value,
                 window.projectEnv,
-                localSuccessFunc);
+                localSuccessFunc,
+            );
         }
     });
 }
 
 function saveProjectBase(path, projectName, mode, successFunc) {
     if (!signedIn()) {
         sweetAlert('Oops!', 'You must sign in to save files.', 'error');
-        updateUI();
-        return;
+
+        return updateUI();
```

Comment:
What's going on here?  `updateUI` doesn't have a return value, does it?

Reply:
No, it doesn't have a return value, it just performs side effects. The intention behind this is to reduce the amount of lines as `return` is simply used to interrupt function execution, hence anything returned will be ignored anyway. I realize it can be confusing - should I revert this and other similar changes?